### PR TITLE
Dcompute fixes

### PR DESCRIPTION
**Pull request by thewilsonator**

### CODE REVIEW

File: gen/dibuilder.cpp [hunk 1]
```diff
@@ -77,12 +77,18 @@ llvm::StringRef uniqueIdent(Type* t) {
 
 
 bool ldc::DIBuilder::mustEmitFullDebugInfo() {
-  // only for -g and -gc
+  // only for -g and -gc but not dcompute (yet)
```
Comment (JohanEngelen) [inline gen/dibuilder.cpp]: best to add "TODO" here so it's findable  (same below)

Comment (thewilsonator) [inline gen/dibuilder.cpp]: Good Idea.

Comment (JohanEngelen) [inline gen/dibuilder.cpp]: and you should add a testcase that passes `-g` together with dcompute flags, testing that it doesn't break compilation, right?

**Comment by JohanEngelen** [comment]
What is this fixing?  (looking at the fixes, I would expect 3 testcases)

**Comment by thewilsonator** [comment]
Will do, I've got to disappear a typeid somewhere first.

**Comment by thewilsonator** [comment]
Added tests.

**Pull request merged**